enhancements/cvo/cluster-profiles.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Cluster Profiles
+authors:
+  - "@csrwng"
+reviewers:
+  - "@abhinavdahiya"
+  - "@crawford"
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-02-04
+last-updated: 2019-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# cluster-profiles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Cluster profiles are a way to support different deployment models for OpenShift clusters. 
+A profile is an identifier that the Cluster Version Operator uses to determine
+which manifests to apply. Operators can be excluded completely or can have different
+manifests for each supported profile.
+
+## Motivation
+
+To support different a deployment model in which not all operators rendered by
+the CVO by default are needed. This includes IBM Public Cloud, in which a 
+hosted control plane is used. Potentially it can also be used for Code Ready Containers.
+
+### Goals
+
+- Equip the CVO to include/exclude manifests based on a matching identifier (profile)
+
+### Non-Goals
+
+- Define which manifests should be excluded/included for each use case
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a user, I can create a cluster in which manifests for control plane operators are
+not applied by the CVO.
+
+#### Story 2
+As a user, I can create a cluster in which node selectors for certain operators target
+worker nodes instead of master nodes.
+
+### Design
+
+A cluster profile is specified to the CVO as an identifier in an environment
+variable. For a given cluster, only one CVO profile may be in effect.
+
+NOTE: The mechanism by which the environment variable is set on the CVO deployment is 
+out of the scope of this design.
+
+```
+CLUSTER_PROFILE=[identifier]
+```
+This environment variable would have to be specified in the CVO deployment. When
+no `CLUSTER_PROFILE=[identifier]` variable is specified, the `default` cluster profile
+is in effect.
+
+The following annotation may be used to include manifests for a given profile:
+
+```
+include.release.openshift.io/[identifier]=true
+```
+This would make the CVO render this manifest only when `CLUSTER_PROFILE=[identifier]`
+has been specified. 
```

Comment:
Probably add a follow-up paragraph that explicitly says manifests with _no_ 
`include.release.openshift.io/*` annotations will always be included.

Reply:
This would be additional information to clarify, but would this hold back merging?

Reply:
Adding text like:

> Manifests with no annotations prefixed by `include.release.openshift.io/` are always included.   Manifests with annotations prefixed by `include.release.openshift.io/` with values other than `true` are undefined.

would cover my concerns (with undefined in [the nasal demons sense][1]).

[1]: https://en.m.wikipedia.org/wiki/Undefined_behavior